src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -357,5 +397,69 @@ public JSONObject getDebugState()
      *
      * @param newVideoConstraints
      */
-    public abstract void setVideoConstraints(Map<String, VideoConstraints> newVideoConstraints);
+    public abstract void setSenderVideoConstraints(ImmutableMap<String, VideoConstraints> newVideoConstraints);
+
+    /**
+     * Notifies this instance that the max video constraints that the bridge
+     * needs to receive from this endpoint has changed. Each implementation
+     * handles this notification differently.
+     *
+     * @param maxVideoConstraints the max video constraints that the bridge
+     * needs to receive from this endpoint
+     */
+    protected abstract void
+    onMaxReceiverVideoConstraintsChanged(VideoConstraints maxVideoConstraints);
+
+    /**
+     * The class that is responsible for bookkeeping of who (which receiver) is
+     * viewing what (video constraints). Internally it maintains a map of
+     * receiver endpoint id -> video constraints. When the map changes, it sets
+     * the new map by calling {@link #setReceiverVideoConstraints(ImmutableMap)}.
+     *
+     * The main benefit of having this class (as opposed to spilling the
+     * implementation in AbstractEndpoint) is to keep the endpoint video
+     * constraints API symmetric (i.e. provide setter for send/recv video
```

Comment:
I am confused. Why would having `addReceiver` and `removeReceiver` in `AbstractEndpoint` make the api non-symmetric? Half the "API" is private anyway. I think having that logic in a separate class is nice, but having to use `getReceiverVideoConstraintsBroker()` is unnecessary detail. I was first going to suggest to add `addReceiver` and `removeReceiver` to `AbstractEndpoint`, calling the broker internally, but since the amount of code is not large I think it's best put it all in `AbstractEndpoint`